txpipe/test/test_utils.py

```diff
@@ -0,0 +1,34 @@
+from ..utils.misc import unique_list, hex_escape, multi_where
+import numpy as np
+
+def test_escape():
+    assert hex_escape(chr(1))=='\\x01'
+    assert hex_escape(chr(7) + chr(11) + chr(12))=='\\x07\\x0b\\x0c'
+
+    assert hex_escape("aaa\nbbb") == "aaa\nbbb"
+    assert hex_escape("xxx\nyyy", replace_newlines=True) == "xxx\\x0ayyy"
+
+def test_unique_list():
+
+    x = [1, 1, 1, 1]
+    assert unique_list(x) == [1]
+
+    x = [1, 1, 2, 1, 1]
+    assert unique_list(x) == [1, 2]
+
+    x = [1, 1, 2, 1, 1, 2, 3, 3]
+    assert unique_list(x) == [1, 2, 3]
+
+    x = [-1, "cat", -1, "cat", "dog"]
+    assert unique_list(x) == [-1, "cat", "dog"]
```

Comment:
This is equivalent to `np.unique(x).tolist()`.  Not sure what the advantage of having an explicitly list-based version would be.  Most of the time `np.unique(x)` will do what you want, and if you absolutely need a list, adding `tolist()` is pretty straightforward.

Reply:
np.unique also sorts its outputs, which I didn't want for the use case of this - I wanted to maintain the order that some results are generated in but remove duplicates.  I couldn't see a numpy equivalent, without the sorting.

Reply:
A quick check shows the time of the two to be pretty close, though both minuscule

```
from txpipe.utils import unique_list
import numpy as np
import time

# use case is tags for data points in sacc
# make array or random 4 letter words with a, b, c, d
letters = np.array(["a", "b", "c", "d"])
words = [np.random.choice(letters, size=4) for i in range(1000)]
words = np.array([''.join(w) for w in words])

# numpy version
t1 = time.time()
_, inds = np.unique(words, return_index=True)
inds.sort()
a = words[inds]

t2 = time.time()

b = unique_list(words)
t3 = time.time()
print('unique_list time = ',t3-t2)
print('numpy time = ',t2-t1)
```

Reply:
OK.  I didn't look at the doc string of unique_list, but you should mention that it preserves order, unlike np.unique, if it doesn't already say so.

FWIW, here is a numpy issue requesting this functionality: https://github.com/numpy/numpy/issues/8621

Reply:
Yep, it mentions it in the docstring.